feat: create frontend coding exercise

bcatoip opened the pull request:
The first file should be a screenshot of what the "completed" exercise should look like. I've removed three bits of code and noted the locations with TODO's. I've also hard-coded the search results since I can sometimes get the pdf to load but usually apryse crashes.

So there are two reasons to post a review here.

1. I want to make sure we're ok with releasing this code to a candidate. It's lifted from our codebase and then heavily modified. I've made this a public repo. But I guess there are probably ways to zip it that might make more sense? I've run this by Chris and he's ok with it.
2. Give you a chance to weigh in on the idea/format/extent of the exercise.

bcatoip committed:
feat: create frontend coding exercise

petejodo commented:
wanted to give some feedback since interview is tomorrow. Overall this looks good but you may want to use something else as a marker other than a `TODO` since there's TODOs that go copied over from our original code. Or maybe remove the unrelated TODOs. Something so that the candidate isn't confused by them. I already moved away from the changes so lost the exact spots but for example I saw a comment like "TODO this should be done in the instance provider".

I think this is a good exercise that we can iterate on. Depending on the candidate does, we could probably simplify some things here

bcatoip committed:
fix: remove some todo's; rename the others

bcatoip commented:
> wanted to give some feedback since interview is tomorrow. Overall this looks good but you may want to use something else as a marker other than a `TODO` since there's TODOs that go copied over from our original code. Or maybe remove the unrelated TODOs. Something so that the candidate isn't confused by them. I already moved away from the changes so lost the exact spots but for example I saw a comment like "TODO this should be done in the instance provider".
> 
> I think this is a good exercise that we can iterate on. Depending on the candidate does, we could probably simplify some things here

I removed the excess TODO's in our code and renamed the ones that were left to "Not implemented". Funny enough, there are a bunch of "Not implemented" comments in the webviewer code too . . . I'm sure it'll be fine.

Yeah, there's definitely room for iteration here. But I was trying to be fairly strict about time-boxing this to a half a day's work on the first pass. My philosophy here is to create something a little more complicated, that requires a bit of context. In other words, something to encourage back and forth between us and the candidate to see if they know how to ask for help and can act on that advice. But I am concerned that I erred on the wrong side of hard.
